### ISSUE TITLE
video tags don't play on mobile

### ISSUE THREAD

**Issue by steff2632** [feederbox826/plugins#14]
like the title says.

Tried on both iOS and android
![IMG_1528](https://github.com/user-attachments/assets/6e32b11c-e5ad-403a-8fb5-b2620e9f8909)

![Screenshot_20241111_204313_Samsung Internet](https://github.com/user-attachments/assets/98d1ee3e-2667-4c13-bb25-19e3e07950b7)

**Comment by feederbox826** [comment]
I was able to replicate it, but they seem like they start playing after a little bit, if you let it sit for a bit, are they all frozen?

**Comment by steff2632** [comment]
turns out this is related to the audio.

A lot of browsers seem to ignore auto playing video's If they're not muted.

An enhancement to this would be to set the covers of the video tag with the thumbs you've already created

**Comment by feederbox826** [comment]
Ah yes, the dreaded autoplay.

> You probably interacted withe the page, which seems to be the trigger for allowing videos which aren't muted to play?

Yes, apparently devtools bypasses it by passing through the mouse, and the second page was slowly loading

> An enhancement to this would be to set the covers of the video tag with the thumbs you've already created

This is an interesting proposition, but stash doesnt support multiple tag images and there's no way to detect blocked autoplay iirc. Posters are currently set as fallback as well for images

**Comment by feederbox826** [comment]
fetching remotely is a possibility, but not one I really want to consider lightly. I'll see if it's possible to inject more blobs into stash

**Comment by steff2632** [comment]
yeah the additional blobs is what i was thinking.

e.g. Gym has a .webm and a .webp so for the `video.poster` attribute you could load the .webp while the .webm loads?

**Comment by feederbox826** [comment]
I couldn't get it to repro on my phone, even without user interaction, this might just have to be the unhappy medium. Right now the <poster> attribute is reserved for image fallbacks as there's no other way to fall back to images if the video fails to load, and stash doesn't expose the mimetype. The last alternative is to laod it all remotely directly from my server or figure out how to effectively sync with python

**Comment by steff2632** [comment]
yep, not sure what the issue is, on my Samsung they load but on the iPhone no, safari desktop also takes an age to load them too.

Had a look at setting the poster for the video tag but the data structure is all obfuscated so will need to pull the db to understand how tags match up with the files in config/blobs

**Comment by feederbox826** [comment]
> Had a look at setting the poster for the video tag but the data structure is all obfuscated so will need to pull the db to understand how tags match up with the files in config/blobs

it doesn't even really help, since we can't see what format it is until it's downloaded (image or video), it becomes

- double request to pull mime type and set poster/video accordingly (doubling latency)
- set up local server to serve tags with a sync job and check for webp/webms
- pull all assets from remote

**Comment by feederbox826** [comment]
I'm becoming more and more interested in this idea, I'll spin it off into it's own repo

**Comment by feederbox826** [comment]
dug into it more and discovered playsInline was not properly set, https://github.com/feederbox826/plugins/commit/919af80ed1596e87713d92ece4c8acd9d413626b fixes the issue for me on chrome mobile, safari **should** allow muted autoplay

**Comment by steff2632** [comment]
this broke the tags for me. Android and iOS.

**Comment by feederbox826** [comment]
reverted error handling, will grab an iphone to test

**Comment by feederbox826** [comment]
Found the problem with safari - if the poster fails to load, the entire media query is cancelled. I'll add in an asset folder for a loading gif and throw in an option since I have forbiddenConfig anyways. Looked into doing remote lookup but it requires the name to be exact

**Comment by feederbox826** [comment]
alternative-tag-layout is still kinda broken, but that's seperate

**Comment by feederbox826** [comment]
https://github.com/feederbox826/plugins/commit/213f2b33ec47d9e00ab06d24ea10722069e7a112

**Comment by steff2632** [comment]
thats worked. Nice one! Well for android, iOS/Safari is still broken but that seems to be just an iOS thing. Will check with developer tools when i get a chance on desktop

**Comment by feederbox826** [comment]
It was working fine for me on iOS safari, just make sure you're ot in low power mode as that will block autoplay no matter what

**Comment by steff2632** [comment]
the images work but the videos don't play after the spinning thing. Work on Android though and on desktop just not my 11 Pro
![IMG_1529](https://github.com/user-attachments/assets/70ecc1b5-7e6a-42a6-9a58-2fe1a11cb834)

**Comment by feederbox826** [comment]
That question mark means format not supported I think

**Comment by feederbox826** [comment]
I'll try again

**Comment by steff2632** [comment]
I tried playing the .webmaster natively and they worked in the browser(although not inline) however it did struggle to load the tags homepage

**Comment by feederbox826** [comment]
I can't seem to figure it out, they play perfectly fine on the iPhone I'm using to test

iOS 18.1.1, iPhone 12 Pro Max. Homepage, within tag... I also have an iPad at my disposal to test...

**Comment by feederbox826** [comment]
Long shot, I got myself an iPhone and could replicate it, still couldn't get it to replicate on iPad but toggling Accessibility -> Motion -> Auto-Play Video Previews on/off fixed it on my end
